src/Event/FileProcessListener.php

```diff
@@ -30,4 +49,38 @@ public function preUpdate(File $file, LifecycleEventArgs $args): void
             }
         }
     }
+
+    /**
+     * Post update method for file entity, to send zipper message when fileset is done.
+     *
+     * @param File               $file An instance of File entity.
+     * @param LifecycleEventArgs $args Doctrine life cycle event args.
+     *
+     * @return void
+     */
+    
+    public function postUpdate(File $file, LifecycleEventArgs $args): void
+    {
+        if (!$file instanceof File) {
+            return;
+        }
+
+        $fileset = $file->getFileset();
+        $datasetSubmission = $fileset->getDatasetSubmission();
+
+        if (!$datasetSubmission instanceof DatasetSubmission) {
+            return;
+        }
+
+        if ($fileset->isDone()) {
```

Comment:
Since this is getting called with every single file, this method is going to get called a lot, I worry a little about the efficiency of the criteria call, but I guess we can optimize it later if it becomes an issue.

Reply:
This call is being resolved in memory, 

> I'm assuming y'all decided not to wait for dataset review to be accepted before generation of the zipfile, because that's how it's currently working. Approving contingent on that being the intent.

No, approval is not a factor in zipping.